src/integrationTest/java/org/opensearch/security/api/RollbackVersionApiIntegrationTest.java

```diff
@@ -0,0 +1,160 @@
+                    + "      }"
+                    + "    }"
+                    + "  },"
+                    + "  {"
+                    + "    \"version_id\": \"v2\","
+                    + "    \"timestamp\": \"2025-04-04T00:00:00Z\","
+                    + "    \"modified_by\": \"admin\","
+                    + "    \"security_configs\": {"
+                    + "      \"internalusers\": {"
+                    + "        \"lastUpdated\": \"2025-04-04T00:00:00Z\","
+                    + "        \"configData\": { \"admin\": { \"hash\": \"$2y$12$bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb\" } }"
+                    + "      }"
+                    + "    }"
+                    + "  }"
+                    + "] }\n";
+
+            var response = client.postJson("/_bulk?refresh=true", bulkPayload);
+            assertThat("Bulk insert failed", response.getStatusCode(), is(200));
+        }
+    }
+
+    @Test
+    public void testRollbackToPreviousVersion_success() throws Exception {
+        withUser(ADMIN_USER_NAME, DEFAULT_PASSWORD, client -> {
+            var response = client.post(ROLLBACK_BASE);
+            assertThat(response.getStatusCode(), is(HttpStatus.SC_OK));
+            assertThat(response.getTextFromJsonBody("/status"), equalTo("OK"));
+            assertThat(response.getTextFromJsonBody("/message"), containsString("config rolled back to version"));
+        });
+    }
+
+    @Test
+    public void testRollbackToSpecificVersion_success() throws Exception {
+        String versionId = "v1";
+        withUser(ADMIN_USER_NAME, DEFAULT_PASSWORD, client -> {
+            var response = client.post(RollbackVersion(versionId));
+            assertThat(response.getStatusCode(), is(HttpStatus.SC_OK));
+            assertThat(response.getTextFromJsonBody("/status"), equalTo("OK"));
+            assertThat(response.getTextFromJsonBody("/message"), containsString("config rolled back to version " + versionId));
+        });
+    }
+
+    @Test
+    public void testRollbackWithNonAdmin_shouldBeUnauthorized() throws Exception {
+        withUser(NEW_USER, DEFAULT_PASSWORD, client -> {
+            var response = client.post(ROLLBACK_BASE);
+            assertThat(response.getStatusCode(), isOneOf(HttpStatus.SC_FORBIDDEN, HttpStatus.SC_UNAUTHORIZED));
+        });
+    }
+
+    @Test
+    public void testRollbackToInvalidVersion_shouldReturnNotFound() throws Exception {
+        withUser(ADMIN_USER_NAME, DEFAULT_PASSWORD, client -> {
+            var response = client.post(RollbackVersion("does-not-exist"));
+            assertThat(response.getStatusCode(), is(HttpStatus.SC_NOT_FOUND));
+            assertThat(response.getTextFromJsonBody("/message"), containsString("not found"));
+        });
+    }
+
+    @Test
+    public void testRollbackWhenOnlyOneVersion_shouldFail() throws Exception {
+        withUser(ADMIN_USER_NAME, DEFAULT_PASSWORD, client -> {
+            client.delete("/.opensearch_security_config_versions");
```

Comment:
A few issues with this:

- I know `.opensearch_security_config_versions` is not formally designated a system index, but if it is in the future then this test will start failing
- On fresh cluster wouldn't there only be one version? Why do we need to delete and insert a dummy value?